xbmc/addons/kodi-addon-dev-kit/include/kodi/addon-instance/PVR.h

```diff
@@ -0,0 +1,3076 @@
+  {
+    std::vector<PVRChannelGroupMember> members;
+    PVR_ERROR error = static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+                          ->GetChannelGroupMembers(group, members);
+    for (auto iter : members)
+    {
+      instance->toKodi->TransferChannelGroupMember(instance->toKodi->kodiInstance, handle, iter);
+    }
+    return error;
+  }
+
+  //--==----==----==----==----==----==----==----==----==----==----==----==----==
+
+  inline static PVR_ERROR ADDON_DeleteChannel(const AddonInstance_PVR* instance,
+                                              const PVR_CHANNEL* channel)
+  {
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+        ->DeleteChannel(channel);
+  }
+
+  inline static PVR_ERROR ADDON_RenameChannel(const AddonInstance_PVR* instance,
+                                              const PVR_CHANNEL* channel)
+  {
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+        ->RenameChannel(channel);
+  }
+
+  inline static PVR_ERROR ADDON_OpenDialogChannelSettings(const AddonInstance_PVR* instance,
+                                                          const PVR_CHANNEL* channel)
+  {
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+        ->OpenDialogChannelSettings(channel);
+  }
+
+  inline static PVR_ERROR ADDON_OpenDialogChannelAdd(const AddonInstance_PVR* instance,
+                                                     const PVR_CHANNEL* channel)
+  {
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+        ->OpenDialogChannelAdd(channel);
+  }
+
+  inline static PVR_ERROR ADDON_OpenDialogChannelScan(const AddonInstance_PVR* instance)
+  {
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+        ->OpenDialogChannelScan();
+  }
+
+  inline static PVR_ERROR ADDON_CallChannelMenuHook(const AddonInstance_PVR* instance,
+                                                    const PVR_MENUHOOK* menuhook,
+                                                    const PVR_CHANNEL* channel)
+  {
+    return static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+        ->CallChannelMenuHook(menuhook, channel);
+  }
+
+  //--==----==----==----==----==----==----==----==----==----==----==----==----==
+
+  inline static PVR_ERROR ADDON_GetEPGForChannel(const AddonInstance_PVR* instance,
+                                                 ADDON_HANDLE handle,
+                                                 int channelUid,
+                                                 time_t start,
+                                                 time_t end)
+  {
+    std::vector<PVREPGTag> epg;
+    PVR_ERROR error = static_cast<CInstancePVRClient*>(instance->toAddon->addonInstance)
+                          ->GetEPGForChannel(channelUid, start, end, epg);
+    for (auto iter : epg)
```

Comment:
in case of error, I would not call the callback.

Please keep in mind that "error_not_implemented" might actually not be an error. Applies to all other error handling comments I just did.